helm: Add example dashboards as chart options

jcpunk opened the pull request:
With this patch, helm users can enable reporting on their clusters with example grafana dashboards.

These dashboards were copied from `examples/kubernetes/addons/prometheus/files/grafana-dashboards/`

Fixes: #21921

```release-note
Integration of sample dashboards with Helm chart
```

gandro reviewed (commented):
Thanks for the PR! I'll ping some folks more familiar with the Grafana integration to also take a look at this. 

Since you copied the Dashboards from `examples` - I think it would make sense to remove them there and update the docs to the new Helm chart options instead. Otherwise we would have to maintain two copies of the dashboards in the tree.

jcpunk commented on install/kubernetes/cilium/values.yaml:
```diff
@@ -1671,6 +1671,14 @@ prometheus:
   metrics: ~
 
+agent:
+  dashboards:
```
I was thinking this made it more explicit on what is going on.  But it does look cleaner to just have a `dashboards` key.  PR updated.

jcpunk commented on install/kubernetes/cilium/templates/cilium-agent/dashboards-configmap.yaml:
```diff
@@ -0,0 +1,25 @@
+{{- if .Values.agent.dashboards.enabled }}
```
oops, fixed.

gandro commented on install/kubernetes/cilium/values.yaml.tmpl:
```diff
@@ -1668,6 +1668,13 @@ prometheus:
   # ref: https://docs.cilium.io/en/stable/operations/metrics/#exported-metrics
   metrics: ~
 
+dashboards:
```
Could you leave some doc comments here to document the meaning of the flags?

jcpunk replied:
I can't seem to find the grafana documentation that describes the `grafana_dashboard` import configmap....

jcpunk replied:
Thanks for the link!

In theory I've got doc updated with comments and links.

gandro commented on install/kubernetes/cilium/files/hubble/dashboards/hubble-dashboard.json:
```diff
@@ -0,0 +1,3232 @@
+{
```
This dashboard is not used by the Helm template. Is that intentional?

The metrics themselves are exposed by Cilium-Agent as well, but on a different port (which is intentional, since we expect the two metric ports to potentially be consumed by different teams)

jcpunk replied:
I thought https://github.com/cilium/cilium/blob/master/install/kubernetes/cilium/templates/hubble/dashboards-configmap.yaml#L2 was importing this as well.

gandro reviewed (commented):
Thanks! I've left some more feedback

gandro commented on install/kubernetes/cilium/values.yaml.tmpl:
```diff
@@ -891,6 +891,9 @@ hubble:
       # -- Metrics relabeling configs for the ServiceMonitor hubble
       metricRelabelings: ~
     dashboards:
+      # -- Grafana dashboards for hubbel
```
```suggestion
      # -- Grafana dashboards for hubble
```

Interestingly enough it's correctly spelled in `values.yaml`. Maybe run `make -C install/kubernetes` again after fixing this.

gandro reviewed (commented):
Thanks! One last typo

jcpunk committed:
helm: Add example dashboards as chart options

With this patch, helm users can enable reporting
on their clusters with example grafana dashboards.

Fixes: #21921
Fixes: #20354

Signed-off-by: Pat Riehecky <riehecky@fnal.gov>

gandro commented on install/kubernetes/cilium/values.yaml:
```diff
@@ -2029,6 +2042,15 @@ operator:
       # -- Metrics relabeling configs for the ServiceMonitor cilium-operator
       metricRelabelings: ~
+  dashboards:
+    # -- Grafana dashboards for cillium-operator
```
More typos (sorry I didn't catch this yesterday):

```suggestion
    # -- Grafana dashboards for cilium-operator
```

gandro commented on install/kubernetes/cilium/values.yaml:
```diff
@@ -2029,6 +2042,15 @@ operator:
+    # -- Grafana dashboards for cillium-operator
+    # grafana can import dashboards based on the label and value
+    # ref: https://github.com/grafana/helm-charts/tree/main/charts/grafana#sidecar-for-dashboards
+    enabled: false
```
For the operator and hubble, the doc comment is above the `enabled` field. For cilium-agent, it's above the `dashboard` field. I think it's best if they are all consistently above the `dashboard` field.

jcpunk replied:
This look better?

gandro replied:
Yes, thanks!

gandro reviewed (commented):
Thanks! Some more documentation nits, and then this should be good to go I think!

gandro commented on install/kubernetes/cilium/values.yaml:
```diff
@@ -1677,6 +1680,16 @@ prometheus:
   # ref: https://docs.cilium.io/en/stable/observability/metrics/
   metrics: ~
 
+# -- Grafana dashboards for cillium-agent
```
This looks much better, thanks! There is one last typo in the generated files. Please re-run `make -C install/kubernetes` and `make -C Documentation helm-values.rst` again. Thanks!

gandro reviewed (approved):
Thank you!
